pkg/cloud/ratelimit_test.go

```diff
@@ -87,27 +87,17 @@ func TestTickerRateLimiter(t *testing.T) {
 	t.Parallel()
 
 	trl := NewTickerRateLimiter(100, time.Second)
-	start := time.Now()
-	for i := 0; i < 50; i++ {
-		err := trl.Accept(context.Background(), nil)
-		if err != nil {
-			t.Errorf("TickerRateLimiter.Accept = %v, want nil", err)
-		}
-	}
-	elapsed := time.Since(start)
-	if elapsed > time.Second {
-		t.Errorf("TickerRateLimiter.Accept took too long: %v, want <1s", elapsed)
-	}
-	if elapsed < 500*time.Millisecond {
-		t.Errorf("TickerRateLimiter.Accept took too short: %v, want >500ms", elapsed)
+	err := trl.Accept(context.Background(), nil)
```

Comment:
maybe you can reverse the logic, instead checking elapsed time just spin the go routine with trl.Accept for 500ms and check how many request were accepted?

Reply:
you mean, you removed time sensitive check not test right? 
sounds good